omod-common/src/main/java/org/openmrs/module/webservices/rest/web/ConversionUtil.java

```diff
@@ -238,15 +240,15 @@ public static Object convert(Object object, Type toType) throws ConversionExcept
 
 
 			if (toClass.isAssignableFrom(Date.class)) {
-				ParseException pex = null;
+				IllegalArgumentException pex = null;
 				String[] supportedFormats = {"yyyy-MM-dd'T'HH:mm:ss.SSSZ", "yyyy-MM-dd'T'HH:mm:ss.SSS",
 						"yyyy-MM-dd'T'HH:mm:ssZ", "yyyy-MM-dd'T'HH:mm:ssXXX", "yyyy-MM-dd'T'HH:mm:ss", "yyyy-MM-dd HH:mm:ss", "yyyy-MM-dd"};
 				for (int i = 0; i < supportedFormats.length; i++) {
 					try {
-						Date date = new SimpleDateFormat(supportedFormats[i]).parse(string);
+						Date date = DateTime.parse(string, DateTimeFormat.forPattern(supportedFormats[i])).toDate();

```

Comment:
Yeah I tried that. Apparently, 3 tests fail when I do
 `Date date = ISODateTimeFormat.dateTime().parseDateTime(string).toDate();`  There errors are:

1. _convert_shouldReturnFalseOnIncorrectDateFromString()_ throws error `java.lang.IllegalArgumentException: Invalid format: "2012-05-01T00:00:00.000" is too short`

2. _convert_shouldReturnCorrectDateWhenParsingStringHavingTimeZone()_ throws error `java.lang.IllegalArgumentException: Invalid format: "2016-01-12T06:00:00+05:30" is malformed at "+05:30"`

3. _convert_shouldReturnEqualsDateFromString()_ throws error `java.lang.IllegalArgumentException: Invalid format: "2011-05-01" is too short`
<br>
Hence I decided to the accepted formats given in the string and parse them the same way it was being done before. @djazayeri